### PR TITLE
feat: GuildMember Class, Roles

### PR DESCRIPTION
**Pull request by Kurpp**
Everything is functional but I believe it could've been done better, open for opinions

### CODE REVIEW

File: src/tests/index.ts [hunk 1]
```diff
@@ -21,6 +21,27 @@ client.on('message', (message: Message) => {
...
 		});
+
+	const clean = (text: unknown) => {
+		if (typeof text === 'string') {
+			return text.replace(/`/g, `\`${String.fromCharCode(8203)}`).replace(/@/g, `@${String.fromCharCode(8203)}`);
+		}
+		return text;
+	};
+
+	if (message.content.startsWith('eval')) {
```
[the remark this answers is not in the document]
Comment (Kurpp) [reply src/tests/index.ts:32]: No specific reason, i was using it just to test role functionality, ill remove it

File: src/structures/GuildMember.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import type { APIGuildMember } from 'discord-api-types/v10';
+import type { Client } from '../client';
+import type { Guild } from './Guild';
+import type { User } from './User';
+import { Base } from './Base';
+
+export class GuildMember extends Base {
+	public readonly id = this.user.id;
+
+	public nick = this.raw.nick;
```
[the remark this answers is not in the document]
Comment (Kurpp) [reply src/structures/GuildMember.ts:10]: I also believe so, but I feel like it should remain as `nick` to be as close to 1:1 of the api, thoughts @Nciklol @twisttaan?

Comment (Nciklol) [inline src/structures/GuildMember.ts:10]:
```suggestion
	public nick?: string = this.raw.nick ?? null;
```

We should be ensuring that there is always *some* value here.

**Review by Nciklol — APPROVED** [review]

**Review by twisttaan — APPROVED** [review]
Good job Neko! Nice to see that you followed our `private raw` flow as well when creating the GuildMember and Role class.

**Comment by Kurpp** [comment]
> Good job Neko! Nice to see that you followed our `private raw` flow as well when creating the GuildMember and Role class.

whos neko 🤔

**Comment by twisttaan** [comment]
> > Good job Neko! Nice to see that you followed our `private raw` flow as well when creating the GuildMember and Role class.
> 
> whos neko 🤔

Your original name, kurpp doesn't have the same charm 😔

**Pull request merged**